### PR TITLE
Make pipeline deletion work

### PR DESCRIPTION
**Pull request by derekchiang**

### CODE REVIEW

**Comment by derekchiang** [comment]
Fixes #281

**Comment by derekchiang** [comment]
@jdoliner plz review

File: src/server/pachyderm_test.go [hunk 1]
```diff
@@ -297,6 +298,36 @@ func TestPipeline(t *testing.T) {
...
+	commit3, err := pfsclient.StartCommit(pachClient, dataRepo, commit2.ID, "")
+	require.NoError(t, err)
+	_, err = pfsclient.PutFile(pachClient, dataRepo, commit3.ID, "file", strings.NewReader("buzz\n"))
+	require.NoError(t, err)
+	require.NoError(t, pfsclient.FinishCommit(pachClient, dataRepo, commit3.ID))
+
+	// We will sleep a while to wait for the pipeline to actually get cancelled
+	// Also if the pipeline didn't get cancelled (due to a bug), we sleep a while
+	// to let the pipeline commit
+	time.Sleep(5 * time.Second)
```
Comment (jdoliner) [inline src/server/pachyderm_test.go]: Bummer that we have to do a sleep here... but I guess there's no way to wait for something to not happen.

**Comment by jdoliner** [comment]
1 very minor comment about an error, other than that everything LGTM

File: src/server/pps/persist/server/rethink_api_server.go [hunk 2]
```diff
@@ -354,14 +359,25 @@ func (a *rethinkAPIServer) SubscribePipelineInfos(request *persist.SubscribePipe
...
+			server.Send(&persist.PipelineInfoChange{
+				Pipeline: change.NewVal,
+			})
+		} else if change.OldVal != nil {
+			server.Send(&persist.PipelineInfoChange{
+				Pipeline: change.OldVal,
+				Removed:  true,
+			})
+		} else {
+			return fmt.Errorf("neither old_val nor new_val was presented in the changefeed; this is likely a bug")
```
Comment (derekchiang) [inline src/server/pps/persist/server/rethink_api_server.go]: We need an english linter

**Pull request merged**